### PR TITLE
Flux Monitor: more explicit timer/ticker configuration and behavior

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/web/testdata/flux_monitor_bridge_job.json [hunk 1]
```diff
@@ -7,8 +7,14 @@
...
       },
       "feeds": [ "https://lambda.staging.devnet.tools/bnc/call", { "bridge": "testbridge" } ],
+      "precision": 2,
       "threshold": 0.5,
-      "precision": 2
+      "idleTimer": {
+        "disabled": true
+      },
+      "pollTimer": {
+        "frequency": "1m"
```
Comment (coventry) [inline core/web/testdata/flux_monitor_bridge_job.json:16]: This is the period, not the frequency.

File: core/services/validators.go [hunk 2]
```diff
@@ -120,28 +120,50 @@ func ValidateInitiator(i models.Initiator, j models.JobSpec, store *store.Store)
...
+	}
+
+	if i.PollTimer.Disabled {
+		if !i.PollTimer.Frequency.IsInstant() {
+			fe.Add("pollTimer disabled, frequency must be 0")
+		}
+	} else {
+		minimumPollFrequency := models.Duration(store.Config.DefaultHTTPTimeout())
+
+		if i.PollTimer.Frequency.IsInstant() {
```
Comment (coventry) [inline core/services/validators.go:148]: `Frequency` should be `Period` or `Duration`.

File: core/store/models/job_spec.go [hunk 3]
```diff
@@ -216,43 +215,22 @@ type InitiatorParams struct {
...
-// defaults represents a default value for an initiator parameter. Value should
-// be the default value, and isZero should be a predicate which returns true if
-// the value of an initiator passed to SetDefaultValues should be overwritten
-// with the default value. See FluxMonitorDefaultInitiatorParams.
-type defaults struct {
-	Value  interface{}
-	isZero func(interface{}) bool
+type PollTimerConfig struct {
+	Disabled  bool     `json:"disabled,omitempty"`
+	Frequency Duration `json:"frequency,omitempty"`
```
Comment (coventry) [inline core/store/models/job_spec.go:228]: If you change this to `Duration`, you can just have a `TimerConfig`.

File: core/services/fluxmonitor/flux_monitor.go [hunk 4]
```diff
@@ -833,44 +786,72 @@ func (p *PollingDeviationChecker) roundState() (contracts.FluxAggregatorRoundSta
...
 			p.roundTimer = nil
-			logger.Debugw("NOT updating roundState.TimesOutAt, negative duration",
-				"value", roundState.TimesOutAt(),
-				"pollDelay", p.pollTicker.d,
-				"idleThreshold", p.idleThreshold,
-				"mostRecentSubmittedRoundID", p.mostRecentSubmittedRoundID,
-				"reportableRoundID", p.reportableRoundID,
-				"contract", p.initr.InitiatorParams.Address.Hex(),
-			)
+			logger.Debugw("disabling roundTimer, negative duration", loggerFields...)
```
Comment (coventry) [inline core/services/fluxmonitor/flux_monitor.go:817]: "roundTimer has run down; disabling" would be clearer. "negative duration" is kind of an implementation detail.

File: core/store/models/job_spec.go (hunk 3 above)
Comment (se3000) [reply core/store/models/job_spec.go:228]: I think that will cause problems with gorm, it won't know how to differentiate columns for `PollTimer` and `IdleTimer`.

File: core/store/models/job_spec.go [hunk 5]
```diff
@@ -218,43 +217,74 @@ type InitiatorParams struct {
...
+	b, err := json.Marshal(ptc)
+	if err != nil {
+		return nil, err
+	}
+	return b, err
+}
+
+// Scan is defined so that we can read PollTimerConfig as JSONB, because
+// of an error with GORM where it has trouble with nested structs as JSONB.
+// See https://github.com/jinzhu/gorm/issues/2704
```
Comment (coventry) [inline core/store/models/job_spec.go:246]: I'm not sure it's an issue with nested structs, I think you may always have to do that. E.g., from the comments on that PR:

> For gorm, implementing driver.Scanner is the only way to mark fields of type struct as normal

**Pull request merged**